src/viking.js

```diff
@@ -1,11 +1,97 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+    this.health = health;
+    this.strength = strength;
+};
+
+Soldier.prototype.attack = function () { 
+    return this.strength;
+};
+
+Soldier.prototype.receiveDamage = function (damage) {
+    if (damage) {
+        this.health -= damage;
+    }
+};
+
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+    Soldier.call(this, health, strength);
+    this.name = name;
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
+
+Viking.prototype.receiveDamage = function (damage) {
+    this.health -= damage; 
+    if (this.health <= 0){
+            return `${this.name} has died in act of combat`
+    }
+    else if (this.health > 0) {
+            return `${this.name} has received ${damage} points of damage`
+    }
+};
+
+Viking.prototype.battleCry = function () {
+    return "Odin Owns You All!";
+};
+
+
 
 // Saxon
-function Saxon() {}
+function Saxon(health, strength) {
+    Soldier.call(this, health, strength);
+};
+
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.constructor = Saxon;
+
+Saxon.prototype.receiveDamage = function (damage) {
+    this.health -= damage; 
+    if (this.health <= 0){
+            return `A Saxon has died in combat`
+    }
+    else if (this.health > 0) {
+            return `A Saxon has received ${damage} points of damage`
+    }
+};
+
 
 // War
-function War() {}
+function War() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+    
+};
+
+War.prototype.addViking = function (viking) {
+   this.vikingArmy.push(viking);
+};
+
+War.prototype.addSaxon = function (saxon) {
+    this.saxonArmy.push(saxon);
+};
+
+War.prototype.vikingAttack = function (){
+    let attackDamage = Viking.strength;
+    this.saxonArmy.pop(saxon);
+    return Saxon.receiveDamage(attackDamage);
+};
```

Comment:
You need to choose a random viking from the viking army and a random saxon from the saxon army.
Once you have them, you apply them the viking.attack() and the saxon.receiveDamage() methods.
You have to do the same for the saxon attack and then, it should solve the problem :)